plugins/govppmux/govpp_channel.go

```diff
@@ -0,0 +1,122 @@
+package govppmux
+
+import (
+	"time"
+
+	govppapi "git.fd.io/govpp.git/api"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging/logrus"
+)
+
+const defaultRetryRequestTimeout = 500 * time.Millisecond
+
+// goVppChan implements govpp channel interface. Instance is returned by NewAPIChannel() or NewAPIChannelBuffered(),
+// and contains *govpp.channel dynamic type (vppChan field). Implemented methods allow custom handling of low-level
+// govpp.
+type goVppChan struct {
+	vppChan govppapi.Channel
+	// Retry data
+	retry *retryConfig
+}
+
+// govppRequestCtx is custom govpp RequestCtx.
+type govppRequestCtx struct {
+	// Original request context
+	requestCtx govppapi.RequestCtx
+	// Function allowing to re-send request in case it's granted by the config file
+	sendRequest func(govppapi.Message) govppapi.RequestCtx
+	// Parameter for sendRequest
+	requestMsg govppapi.Message
+	// Retry data
+	retry *retryConfig
+}
+
+// helper struct holding info about retry configuration
+type retryConfig struct {
+	attempts int
+	timeout  time.Duration
+}
+
+// ReceiveReply handles request and returns error if occurred. Also does retry if this option is available.
+func (r *govppRequestCtx) ReceiveReply(reply govppapi.Message) error {
+	var err error
+	// Receive reply from original send
+	if err = r.requestCtx.ReceiveReply(reply); err != nil && err == core.ErrNotConnected {
+		if r.retry != nil && r.retry.attempts > 0 {
```

Comment:
These values should probably be kept as local variables for the scope of `ReceiveReply`, to prevent changes that might occur during retries.

Reply:
good point